components/org.wso2.carbon.deployment.engine/src/test/java/org/wso2/carbon/deployment/engine/deployers/CustomDeployer.java

```diff
@@ -93,16 +105,24 @@ public void undeploy(Object key) throws CarbonDeploymentException {
                     "is not a String value");
         }
         logger.info("Undeploying : " + key);
-        try {
-            File fileToUndeploy = new File(testDir + File.separator + key);
-            logger.info("File to undeploy : " + fileToUndeploy.getAbsolutePath());
-            FileInputStream fis = new FileInputStream(fileToUndeploy);
+        File fileToUndeploy;
+        if (key.equals("sample1.txt")) {
+            fileToUndeploy = new File(testDir + File.separator + key);
+        } else if (key.equals("sample2.txt")) {
+            fileToUndeploy = new File(testDir2 + File.separator + key);
+        } else {
+            throw new CarbonDeploymentException("Error while Un Deploying : " + key);
+        }
+        logger.info("File to undeploy : " + fileToUndeploy.getAbsolutePath());
+        try (FileInputStream fis = new FileInputStream(fileToUndeploy)) {
             int x = fis.available();
             byte b[] = new byte[x];
             fis.read(b);
             String content = new String(b);
             if (content.contains("sample1")) {
                 sample1Deployed = false;
+            } else if (content.contains("sample2")) {
```

Comment:
Shall we turn-around the equal check for avoiding any NPEs ?
For ex: "something".equals(string)